src/App.jsx

```diff
@@ -14,45 +16,124 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+function getPrepearedGoods(goods, { currentSortField, reversedState }) {
+  const prepearedGoods = [...goods];
+
+  if (currentSortField === '') {
+    if (reversedState === 'reversed') {
+      prepearedGoods.reverse();
+    }
+  }
+
+  if (currentSortField === 'Name') {
+    if (reversedState === 'reversed') {
+      prepearedGoods.sort().reverse();
+    }
+
+    if (reversedState !== 'reversed') {
+      prepearedGoods.sort();
+    }
+  }
+
+  if (currentSortField === 'Length') {
+    if (reversedState === 'reversed') {
+      prepearedGoods.sort(
+        (good1, good2) => (good1.length - good2.length),
+      ).reverse();
+    }
+
+    if (reversedState !== 'reversed') {
+      prepearedGoods.sort((good1, good2) => (good1.length - good2.length));
+    }
+  }
+
+  return prepearedGoods;
+}
+
+export const App = () => {
+  const [currentSortField, setCurrentSortField] = useState('');
+  const [reversedState, setReversedState] = useState('');
+  const vissibleGoods = getPrepearedGoods(goodsFromServer,
+    { currentSortField, reversedState });
+
+  const isStateNotReversedOrSorted
+  = (reversedState === 'reversed') || (currentSortField !== '');
+
+  const sortByName = () => {
+    setCurrentSortField('Name');
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const sortByLength = () => {
+    setCurrentSortField('Length');
+  };
+
+  const reverse = () => {
+    if (reversedState === 'reversed') {
+      setReversedState('');
+    }
+
+    if (reversedState !== 'reversed') {
+      setReversedState('reversed');
+    }
+  };
+
+  const reset = () => {
+    setReversedState('');
+    setCurrentSortField('');
+  };
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          onClick={sortByName}
+          type="button"
+          className={
+            classnames('button is-info',
```

Comment:
`button` and `is-info` are 2 different classes, they should be written in separate ''